endpoints/openrtb2/amp_auction.go

```diff
@@ -143,16 +143,16 @@ func (deps *endpointDeps) AmpAuction(w http.ResponseWriter, r *http.Request, _ h
 	defer cancel()
 
 	usersyncs := usersync.ParsePBSCookieFromRequest(r, &(deps.cfg.HostCookie))
-	if req.App != nil {
-		labels.Source = pbsmetrics.DemandApp
+	labels.Source = pbsmetrics.DemandWeb
+	if usersyncs.LiveSyncCount() == 0 {
+		labels.CookieFlag = pbsmetrics.CookieFlagNo
 	} else {
-		labels.Source = pbsmetrics.DemandWeb
-		if usersyncs.LiveSyncCount() == 0 {
-			labels.CookieFlag = pbsmetrics.CookieFlagNo
-		} else {
-			labels.CookieFlag = pbsmetrics.CookieFlagYes
-		}
+		labels.CookieFlag = pbsmetrics.CookieFlagYes
 	}
+	if req.Site.Publisher != nil && req.Site.Publisher.ID != "" {
```

Comment:
Do we expect Site to be set in an amp request?

Reply:
Yes. The parseAmpRequest function calls `(deps *endpointDeps) validateRequest(req *openrtb.BidRequest)` in line 262. The check in line 303 of the `validateRequest` function, guarantees `req.Site` and `req.App` are not both NULL nor not NULL

Reply:
```
 258 func (deps *endpointDeps) validateRequest(req *openrtb.BidRequest) []error {
 259     errL := []error{}
 260 +-- 42 lines: if req.ID == "" {-----------------------------------------------
 302
 303     if (req.Site == nil && req.App == nil) || (req.Site != nil && req.App != nil) {
 304         errL = append(errL, errors.New("request.site or request.app must be defined, but not both."))
 305         return errL
 306     }
 307
```

Reply:
This is a fair point. The check in line 257, `errL := deps.validateRequest(req)` can return true even if `req.Site` is NULL. Let me put it back in.